### PR TITLE
feat: support type filter in list API and CLI

### PR DESCRIPTION
**Pull request by AlekSi**
# Pull Request

## What? (description)

Support type filter in list API and CLI.

## Why? (reasoning)

#2068.

## Acceptance

Please use the following checklist:

- [x] you linked an issue (if applicable)
- [x] you included tests (if applicable)
- [x] you ran conformance (`make conformance`)
- [x] you formatted your code (`make fmt`)
- [x] you linted your code (`make lint`)
- [x] you generated documentation (`make docs`)
- [x] you ran unit-tests (`make unit-tests`)

> See `make help` for a description of the available targets.

### CODE REVIEW

File: website/content/docs/v0.8/Reference/cli.md [hunk 1]
```diff
@@ -91,11 +91,11 @@ talosctl cluster create [flags]
       --endpoint string                         use endpoint instead of provider defaults
   -p, --exposed-ports string                    Comma-separated list of ports/protocols to expose on init node. Ex -p <hostPort>:<containerPort>/<protocol (tcp or udp)> (Docker provisioner only)
   -h, --help                                    help for create
-      --image string                            the image to use (default "ghcr.io/talos-systems/talos:latest")
+      --image string                            the image to use (default "ghcr.io/aleksi/talos:latest")
```
Comment (AlekSi) [inline website/content/docs/v0.8/Reference/cli.md:94]: Oh :)

Comment (smira) [reply website/content/docs/v0.8/Reference/cli.md:94]: make it more personal :)

**Review by smira — APPROVED** [review]
:+1:

**Comment by smira** [comment]
one thing we could do is to add a test to https://github.com/talos-systems/talos/blob/master/internal/integration/cli/list.go for the new features, but not strictly required, as it's hard to have something predictable on server side we could `talosctl ls` and build some assertion around it

**Review by Unix4ever — APPROVED** [review]
:cool:

**Pull request merged**